code/game/jobs/access.dm

```diff
@@ -277,22 +278,27 @@ proc/get_all_job_icons()
 		if("E4") . = size ? "CPL " : "Corporal"
 		if("E5") . = size ? "SGT " : "Sergeant"
 		if("E6") . = size ? "SSGT " : "Staff Sergeant"
-		if("E6E") . = size ? "TSGT " : "Technical Sergeant"
-		if("E7") . = size ? "SFC " : "Sergeant First Class"
+		if("E7") . = size ? "GYSGT " : "Gunnery Sergeant"
 		if("E8") . = size ? "MSGT " : "Master Sergeant"
 		if("E8E") . = size ? "FSGT " : "First Sergeant"
-		if("E9") . = size ? "SGM " : "Sergeant Major"
-		if("E9E") . = size ? "CSGM " : "Command Sergeant Major"
+		if("E9") . = size ? "SGTMAJ " : "Sergeant Major"
+		if("E9E") . = size ? "CSGTMAJ " : "Command Sergeant Major"
 		if("O1") . = size ? "ENS " : "Ensign"
-		if("O2") . = size ? "LT " : "Lieutenant"
-		if("O3") . = size ? "LCDR " : "Lieutenant Commander"
-		if("O4") . = size ? "CDR " : "Commander"
-		if("O5") . = size ? "CPT " : "Captain"
-		if("O6") . = size ? "RADM " : "Rear Admiral"
-		if("O7") . = size ? "ADM " : "Admiral"
-		if("O8") . = size ? "FADM " : "Fleet Admiral"
-		if("O9") . = size ? "SMR " : "Sky Marshal"
+		if("O2") . = size ? "LTJG " : "Lieutenant Junior Grade"
+		if("O3") . = size ? "LT " : "Lieutenant"
+		if("O4") . = size ? "LCDR " : "Lieutenant Commander"
+		if("O5") . = size ? "CDR " : "Commander"
+		if("O6") . = size ? "CPT " : "Captain"
+		if("O7") . = size ? "COMM " : "Commodore"
+		if("O8") . = size ? "RADM " : "Rear Admiral"
+		if("O9") . = size ? "VADM " : "Vice Admiral"
+		if("10") . = size ? "ADM " : "Admiral"
+		if("11") . = size ? "FADM " : "Fleet Admiral"
+		if("PO") . = size ? "PO " : "Petty Officer"
+		if("CPO") . = size ? "CPO " : "Chief Petty Officer"
 		if("WO") . = size ? "WO " : "Warrant Officer"
+		if("CWO") . = size ? "CWO " : "Chief Warrant Officer"
+		if("MAJ") . = size ? "MAJ " : "Major"
```

Comment:
Could also list the commissioned marine ranks in some way like such as `M-O4`, which is the Major's pay grade.

Reply:
I don't see any other marine officer ranks being used in the near future, and for events they wouldn't really be that useful since marine officers are low on the chain of command, if you want I could add them but I wouldn't unless asked.